### DISCUSSION TITLE
Improve error message for composite primary key failure

### DISCUSSION THREAD

**Discussion by Sjlver** [directus/directus#24896] [Feature Requests]
### Describe the Bug

Directus does not handle tables with composite primary keys very well. For example, consider the following:

```sql
CREATE TABLE Room (
    Floor int,
    RoomNumber int,
    Label nvarchar(max),
    PRIMARY KEY (Floor, RoomNumber)
);
```

When connecting to a database with such a table, Directus prints the following log message:

```
WARN: Collection "Room" doesn't have a primary key column and will be ignored
```

This is somewhat misleading, since there is a primary key -- just not of the type that Directus would expect.

This collection nevertheless shows up under `/admin/settings/data-model/Room`. Admins can successfully set the description. However, none of the fields show up. Trying to add a field results in the message `[FORBIDDEN] You don't have permission to access this.`

Once the collection has been set up (e.g., a description was set), it also shows up under `/admin/content/Room`. Accessing it there shows this message:

![image](https://user-images.githubusercontent.com/72059/235425876-d349608e-861b-4236-8a99-6029cdf1e6c5.png)

---

Overall, this is *not* a feature request to handle such complicated tables. I only think that Directus should fail more gracefully here and provide more correct, user-friendly error messages. Concretely:

- Change the warning to something like `Collection "Room " has a composite primary key column and will be ignored`.
- Under the data model settings, say something like `This collection does not have a primary key that is compatible with Directus.`
- Never show an incompatible collection in the content section.

### To Reproduce

Create the table from the description above in any database, and try to access it using Directus.

### Directus Version

v10.0.0

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by torrybr** [comment]
Experiencing this as well, Is there any roadmap for composite key support?

**Comment by rantoniuk** [comment]
@rijkvanzanten [looking at this discussion](https://github.com/directus/directus/discussions/11897), can you shed some light on the hectic status changes of this issue on the roadmap? I don't understand [why there is no specific issue created for composite key support](https://github.com/orgs/directus/projects/13/views/12?filterQuery=composite) even in the Backlog, after all, composite keys for M2M tables are widely used for years in RDBMS systems.

Having the discussion closed, makes it even impossible to vote it up to express the need per your feature request process - and I don't think a new discussion thread is needed, just voting the old one should be enough.
